tests/test_bidsArchive.py

```diff
@@ -776,6 +826,24 @@ def testGetBidsRun(bidsArchiveMultipleRuns, sampleBidsEntities, sample4DNifti1,
     assert run.numIncrementals() == sample4DNifti1.header.get_data_shape()[3]
 
 
+# Test getBidsRun gets inherited events data
+def testGetBidsRunInheritedEvents(bidsArchive4D, sampleBidsEntities):
```

Comment:
It's not clear to me this is testing inheritance with multiple events files at different levels.

Reply:
I've expanded this test and reworked some of the logic, latest commit shows those changes.